money_trackerV2.py

```
@@ -34,13 +34,13 @@ def transaction(self):
                 except UnboundLocalError:
```

I think it would be better to do bounds checking (such as if-statements to prevent keys or indices that go out of bounds) than to just ignore the exception